src/bookshelf/test/bkswt_api_SUITE.erl

```diff
@@ -0,0 +1,687 @@
+%% -*- mode: Erlang; fill-column: 80; comment-column: 75; -*-
+%%-------------------------------------------------------------------
+%% @author Eric B Merritt <ericbmerritt@gmail.com>
+%% @author Mark Anderson <mark@chef.io>
+%% Copyright 2012-5 Opscode, Inc. All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+
+-module(bkswt_api_SUITE).
+
+%% Note: This directive should only be used in test suites.
+-compile(export_all).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("eunit/include/eunit.hrl").
+-include("../src/internal.hrl").
+
+-define(STANDALONE_BOOKSHELF, false).
+
+-define(STR_CHARS, "abcdefghijklmnopqrstuvwxyz").
+
+%% Loads the environment from a config file
+load_default_config() ->
+    case ?STANDALONE_BOOKSHELF of
+        true ->
+            load_config_from_file();
+        false ->
+            load_inline_config()
+    end.
+
+load_config_from_file() ->
+    ConfigPath = case os:getenv("BOOKSHELF_CT_CONFIG_PATH") of
+                     false -> "/var/opt/opscode/bookshelf/sys.config";
+                     X -> X
+                 end,
+    load_config(ConfigPath).
+
+load_inline_config() ->
+    StaticBookshelfConfig = [{ip, "127.0.0.1"},
+                             {port, 4321},
+                             {keys, {"e1efc99729beb175",
+                                     "fc683cd9ed1990ca"}},
+                             {disk_store, "/tmp/bukkits"},
+                             {sql_retry_count, 5},
```

Comment:
I think we should definitely test with whatever the default is for retries (currently 0 as you mention).  Do we want to make this configurable for end users?  A concern I have is that it isn't applied universally to all sql interactions. cc @manderson26